src/ctia/bulk/core.clj

```diff
@@ -157,6 +157,31 @@
     [:ctia :store :bulk-refresh]
     "false"))
 
+(s/defn create-asset-ref-entities
+  "Creates AssetMapping and AssetProperties entities, rectifying transient IDs in the process"
+  [bulk temp-ids login refresh
+   services :- APIHandlerServices]
+  (let [bulk' (select-keys bulk [:asset_mappings :asset_properties])
+        ;; replace :asset_ref fields with non-transient IDs
+        set-ref (fn [{:keys [asset_ref] :as m}]
+                  (if (schemas/transient-id? asset_ref)
+                    ;; TODO: add the check described in https://github.com/threatgrid/iroh/issues/4917
+                    (assoc m :asset_ref (get temp-ids asset_ref))
+                    m))
+        ents (zipmap
+              (keys bulk')
+              (->> bulk'
+                   vals
+                   (map (partial map set-ref))))]
+    ;; create asset-mapping and asset-properties entities
+    (gen-bulk-from-fn
+     create-entities
+     ents
+     temp-ids
+     login
+     {:refresh refresh}
+     services)))
```

Comment:
Indeed you should simply do this:
``` clojure
(gen-bulk-from-fn
     create-entities
     (select-keys bulk [:asset_mappings :asset_properties])
     temp-ids
     login
     {:refresh refresh}
     services)))
```
and move the transient id resolution  to a realize-fn function that would be defined as a field of asset-mapping and asset-properties

Reply:
done in https://github.com/threatgrid/ctia/pull/1089/commits/261f262171cbfef23ad467ac63eeb5cc1b69b6e2